### PR TITLE
Add test_cpp_extensions tests for stream_and_event and mita_backend

### PR DESCRIPTION
**Pull request by egienvalue**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123614
* #123612
* #123611

Test the generic torch.Stream/Event with fake device gurad and hooks. Since we added a fake device backend, it is mutual exclusive to other backends. Tests will be skipped if TEST_CUDA or TEST_ROCM is true.

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Super cool!

File: test/test_cpp_extensions_stream_and_event.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Owner(s): ["module: unknown"]
```
Comment (malfet) [inline test/test_cpp_extensions_stream_and_event.py:1]: We should not add new unowned tests to CI

File: test/test_cpp_extensions_stream_and_event.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            name="mtia_extension",
+            sources=[src],
+            build_directory=build_dir,
+            extra_include_paths=[
+                "cpp_extensions",
+                "path / with spaces in it",
+                "path with quote'",
+            ],
+            is_python_module=False,
+            verbose=True,
```
Comment (malfet) [inline test/test_cpp_extensions_stream_and_event.py:79]: It feels like this code is verbatim copy from another file, is this really needed (I.e. can both tests perhaps live in the same file?)

Comment (egienvalue) [reply test/test_cpp_extensions_stream_and_event.py:79]: The test_cpp_extension_mtia_backend.py one ? They have different purposes, but they are using same cpp extension. I can create the shared cpp extension @malfet, how do you think ? 

test_cpp_extension_mtia_backend.py -> test torch.mtia module.
test_cpp_extension_stream_and_event.py -> test the general torch.Stream/Event APIs.

File: test/test_cpp_extensions_stream_and_event.py (hunk 1 above)
Comment (albanD) [reply test/test_cpp_extensions_stream_and_event.py:1]: I created a "module: mtia" (https://github.com/pytorch/pytorch/labels/module%3A%20mtia) that you should use there.
Let's discuss offline how to make sure someone is looking at it!

**Comment by egienvalue** [comment]
@egienvalue has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56443358).

**Comment by egienvalue** [comment]
I am relanding the PR. All ROCm signals look good. 
cc @jeffdaily  @jithunnair-amd

**Pull request closed without merging**